File: prog3/trab_cpp/Outro.cpp
#include "Outro.hpp"

Outro::Outro(string natureza, string idioma, string cidade, int qtd_pags, string editora): Producao("", natureza, idioma, cidade, qtd_pags, 10){
    this->editora = editora;
}

Outro::~Outro(){}

/* Metodo que retorna a string que contem as informacoes de uma producao em formato csv. No caso, reorna uma string no seguinte formato:
 "<Natureza>;<Idioma>;<Editora>;<Cidade>;<Paginas>"*/
string Outro::csv(){
    stringstream stream;
    stream << natureza << ";" << idioma << ";" << editora << ";" << cidade << ";" << _int_string(qtd_pags);
    return stream.str();
}

/* Metodo que retorna a string cabecalho. */
string Outro::cabecalho(){
    return "Natureza;Idioma;Editora;Cidade;Paginas";
}

/* Operador de comparacao de uma producao com outra de mesmo subtipo. */
bool Outro::operator < (Producao& o){
    Outro& prod = (Outro&)o;
    int cmp = natureza.compare(prod.natureza);
    if(cmp<0) return true; else if(cmp>0) return false;
    cmp = idioma.compare(prod.idioma);
    if(cmp<0) return true; else if(cmp>0) return false;
    cmp = editora.compare(prod.editora);
    if(cmp<0) return true; else if(cmp>0) return false;
    cmp = cidade.compare(prod.cidade);
    if(cmp<0) return true; else if(cmp>0) return false;
    cmp = qtd_pags - prod.qtd_pags;
    if(cmp<0) return true; else return false;
}

File: prog3/trab_java/utilitarios/capes/producoes/Outro.java
package utilitarios.capes.producoes;

public class Outro extends Producao implements Comparable<Outro> {
    private String editora;

    public Outro(String natureza, String idioma, String editora, String cidade, int qtdPags){
        super(null, natureza, idioma, cidade, qtdPags);
        this.editora = editora;
        this.tipo = 7;
    }

    @Override
    public int compareTo(Outro o) {
        return this.natureza.compareTo(o.natureza) != 0 ?
        this.natureza.compareTo(o.natureza) : this.idioma.compareTo(o.idioma) != 0 ? 
        this.idioma.compareTo(o.idioma) : this.editora.compareTo(o.editora) != 0 ?
        this.editora.compareTo(o.editora) : this.cidade.compareTo(o.cidade) != 0 ?
        this.cidade.compareTo(o.cidade) : this.qtdPags - o.qtdPags;
    }
    
    public static void imprimeCabecalho(){
        System.out.println("Natureza;Idioma;Editora;Cidade;Paginas");
    }

    @Override
    public String toString(){
        return (this.natureza + ";" + this.idioma + ";" + this.editora + ";" + this.cidade + ";" + this.qtdPags_string);
    } 
}